src/test/java/FunctionalTest.java

```diff
@@ -0,0 +1,60 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class FunctionalTest  {
+    @Test
+    void testIfListIsEmptyForSum() {
+        ArrayList<Integer> testList = new ArrayList<>();
+        assertEquals(0, Functional.sum(testList));
+    }
+
+    @Test
+    void testIfListHasSameElements() {
+        int number = 5;
+        assertEquals(number * 3, Functional.sum(Arrays.asList(5, 5, 5)));
+    }
+
+    @Test
+    void commutativeTest() {
+        assertEquals(Functional.sum(Arrays.asList(1, 2, 3)), Functional.sum(Arrays.asList(3, 2, 1)));
+    }
+
+    @Test
+    void associativeTest() {
+        assertEquals(Functional.sum(Arrays.asList(1, 2, 3, 4)), Functional.sum(Arrays.asList(1, 2)) + Functional.sum(Arrays.asList(3, 4)));
```

Comment:
Try to keep lines at most **120** symbols long. For instance, in this case, you could succeed it either extracting sum to the variable or just adding static import of `Functional.sum` and the same for `Functional.max`.

```java
final int sumOfList = Functional.sum(Arrays.asList(1, 2, 3, 4));
final int sumOfSublists = Functional.sum(Arrays.asList(1, 2)) + Functional.sum(Arrays.asList(3, 4));
assertEquals(sumOfList, sumOfSublists);
```

Reply:
I forgot to write an alternative solution... The alternative solution is to import the functions `max` and `sum` statically. But for this, you will need to create a package and move all the `java` files in the package. Structure will be like this:
```
.
└── src
    ├── main
    │   └── java
    │       └── example
    │           └── Functional.java
    └── test
        └── java
            └── example
                └── FunctionalTest.java
```
Add at the very top of the `java` files package:
```java
package example;
```
Then in the test class you can do the following:
```java
package example;

/* Non-static imports */

import static example.Fanctional.sum;
import static example.Fanctional.max;

/* Class definition */
```
And finally you can significally shorten you code:
```java
assertEquals(sum(Arrays.asList(1, 2, 3, 4)), sum(Arrays.asList(1, 2)) + sum(Arrays.asList(3, 4)));
```
And, of course, if you would like you can add static import for `Arrays.asList` also.
```java
assertEquals(sum(asList(1, 2, 3, 4)), sum(asList(1, 2)) + sum(asList(3, 4)));
```